core/state/statedb.go

```diff
@@ -855,11 +863,52 @@ func (s *StateDB) IntermediateRoot(deleteEmptyObjects bool) common.Hash {
 	// the account prefetcher. Instead, let's process all the storage updates
 	// first, giving the account prefeches just a few more milliseconds of time
 	// to pull useful data from disk.
+
+	// ---------------------------------------------- DEBUGGING -------------------------------------------------------------------------------------------
+
+	// Get the stateObjects needed to be updated
+	updateObjs := []*stateObject{}
 	for addr := range s.stateObjectsPending {
 		if obj := s.stateObjects[addr]; !obj.deleted {
-			obj.updateRoot(s.db)
+			updateObjs = append(updateObjs, obj)
+			obj.updateTrie(s.db, true)
 		}
 	}
+	min := func(a, b int) int {
```

Comment:
With the new way of calculating number of objects per routine, this function is only called once, so I think removing this function and using if for that one is better